### PR TITLE
implemented cardCountSection

### PR DESCRIPTION
**Pull request by mkotyak**

### CODE REVIEW

File: Memorize/Screens/ThemeEditor/ThemeEditorView.swift [hunk 1]
```diff
@@ -3,6 +3,22 @@ import SwiftUI
...
+                return "All"
+            } else if pairsValue == .random {
+                return "Random"
+            } else {
+                return "Explicit"
+            }
+        }
+
+        self.theme = theme
+        self.pairsValue = num
```
Comment (garbalinskiy) [inline Memorize/Screens/ThemeEditor/ThemeEditorView.swift]: `num`? what is `num`? )

File: Memorize/Screens/ThemeEditor/ThemeEditorView.swift [hunk 2]
```diff
@@ -3,6 +3,22 @@ import SwiftUI
 struct ThemeEditorView: View {
     var theme: ThemeEditorViewModel
     @State private var emojisToAdd = ""
+    @State var pairsValue: String
+
+    init(theme: ThemeEditorViewModel, pairsValue: ThemeModel.NumOfPairs) {
+        var num: String {
```
Comment (garbalinskiy) [inline Memorize/Screens/ThemeEditor/ThemeEditorView.swift]: this part of code shouldn't be in View

File: Memorize/Screens/ThemeEditor/ThemeEditorView.swift [hunk 3]
```diff
@@ -3,6 +3,22 @@ import SwiftUI
 struct ThemeEditorView: View {
     var theme: ThemeEditorViewModel
```
Comment (garbalinskiy) [inline Memorize/Screens/ThemeEditor/ThemeEditorView.swift]: it seems that this is not `theme` anymore and it's `viewModel`

File: Memorize/Screens/ThemeEditor/ThemeEditorView.swift [hunk 4]
```diff
@@ -45,15 +61,40 @@ struct ThemeEditorView: View {
 
     var cardCountSection: some View {
         Section(header: Text("Card count")) {
-            HStack {
-                Text("Num of pairs")
-                Stepper("") {} onDecrement: {}
+            VStack {
+                HStack {
+                    Picker(selection: $pairsValue) {
```
Comment (garbalinskiy) [inline Memorize/Screens/ThemeEditor/ThemeEditorView.swift:66]: why it's not possible to bind it with `viewModel. pairsValue` ?

File: Memorize/Screens/ThemeEditor/ThemeEditorView.swift [hunk 5]
```diff
@@ -45,15 +61,40 @@ struct ThemeEditorView: View {
...
+                        let numberCases = ["All", "Random", "Explicit"]
```
Comment (garbalinskiy) [inline Memorize/Screens/ThemeEditor/ThemeEditorView.swift:67]: This array should be provided from `ViewModel`

File: Memorize/Screens/ThemeEditor/ThemeEditorView.swift [hunk 6]
```diff
@@ -45,15 +61,40 @@ struct ThemeEditorView: View {
...
+                    .onChange(of: pairsValue) { newValue in
+                        theme.updatePairs(newValue)
+                    }
+                }
+
+                if pairsValue == "Explicit" {
+                    var count = theme.pairsCount()
+
+                    Stepper("\(count) pairs") {
+                        count += 1
```
Comment (garbalinskiy) [inline Memorize/Screens/ThemeEditor/ThemeEditorView.swift:84]: Please try to interact directly with ViewModel

File: Memorize/Screens/ThemeEditor/ThemeEditorViewModel.swift [hunk 7]
```diff
@@ -36,4 +36,37 @@ class ThemeEditorViewModel: ObservableObject {
             theme.emojis.remove(at: index)
         }
     }
+
+    func pairsCount() -> Int {
+        var count = 0
```
Comment (garbalinskiy) [inline Memorize/Screens/ThemeEditor/ThemeEditorViewModel.swift:41]: such approach leads to reset explicit number of pars when to 0 when change form explicit to any other and back

File: Memorize/Screens/ThemeEditor/ThemeEditorViewModel.swift [hunk 8]
```diff
@@ -36,4 +36,37 @@ class ThemeEditorViewModel: ObservableObject {
...
+    }
+
+    func updatePairs(_ value: String) {
+        switch value {
+        case "All":
+            theme.numOfPairs = .all
+        case "Random":
+            theme.numOfPairs = .random
+        case "Explicit":
+            theme.numOfPairs = .explicit(0)
```
Comment (garbalinskiy) [inline Memorize/Screens/ThemeEditor/ThemeEditorViewModel.swift:59]: and here about reset to 0 the same

File: Memorize/Screens/ThemeEditor/ThemeEditorViewModel.swift [hunk 9]
```diff
@@ -36,4 +36,37 @@ class ThemeEditorViewModel: ObservableObject {
...
+        default:
+            break
+        }
+    }
+
+    func incrementPairsCount(_ newValue: Int) {
+        theme.numOfPairs = .explicit(min(emojis.count, newValue))
```
Comment (garbalinskiy) [inline Memorize/Screens/ThemeEditor/ThemeEditorViewModel.swift:66]: this should be enough to update UI without keeping this value in View

**Review by garbalinskiy — APPROVED** [review]

File: Memorize/Screens/ThemeEditor/ThemeEditorView.swift (hunk 1 above)
Comment (mkotyak) [reply Memorize/Screens/ThemeEditor/ThemeEditorView.swift]: Yes, sorry. I tried different approaches and forgot to rename it

File: Memorize/Screens/ThemeEditor/ThemeEditorView.swift (hunk 3 above)
Comment (mkotyak) [reply Memorize/Screens/ThemeEditor/ThemeEditorView.swift]: agree

**Pull request merged**